Source/WebCore/css/parser/CSSParserTokenRange.cpp

```diff
@@ -103,31 +112,31 @@ void CSSParserTokenRange::consumeComponentValue()
             nestingLevel++;
         else if (token.getBlockType() == CSSParserToken::BlockEnd)
             nestingLevel--;
-    } while (nestingLevel && m_first < m_last);
+    } while (nestingLevel && !m_tokens.empty());
 }
 
 void CSSParserTokenRange::trimTrailingWhitespace()
 {
-    for (; m_last > m_first; --m_last) {
-        if (!CSSTokenizer::isWhitespace((m_last - 1)->type()))
-            return;
-    }
+    size_t i = m_tokens.size();
+    for (; i > 0 && CSSTokenizer::isWhitespace(m_tokens[i - 1].type()); --i) { }
+    m_tokens = m_tokens.first(i);
```

Comment:
This could be

```
while (m_tokens.size() && CSSTokenizer::isWhitespace(m_tokens.last())) {
    dropLast(m_tokens)
}
```

(if you added a WTF::dropLast helper function)

Reply:
Hopefully this is identical perf-wise since I A/B tested this. I was trying to update the span only once instead of once per loop iteration but maybe it doesn't matter.

Reply:
I suspect it won't matter because the previous code was happy to update m_last, which is pretty much identical.